Che 7 plugin: PHP Language Server #12797

vrubezhny opened the pull request:
Add PHP Intelephense v.1.0.13 for PHP Landuage Support
From https://marketplace.visualstudio.com/items?itemName=bmewburn.vscode-intelephense-client

Signed-off-by: Victor Rubezhny <vrubezhny@redhat.com>

### What does this PR do?

This PR adds a Che7 plugin for PHP Language support based on PHPIntelephense VSCode extension: https://marketplace.visualstudio.com/items?itemName=bmewburn.vscode-intelephense-client

Fixes eclipse/che#12797

ibuziuk commented on plugins/ms-vscode.php.intelephense/1.0.13/meta.yaml:
```diff
@@ -0,0 +1,14 @@
+category: Language
+firstPublicationDate: "2019-04-16"
+attributes:
+  containerImage: "eclipse/che-remote-plugin-php7:next"
```
no new line in the end

vrubezhny commented:
> reference to the marketplace should be avoided, please ask plugin maintainer to provide vsix as part of the GH release

The issue https://github.com/bmewburn/vscode-intelephense/issues/465 is created for this problem.

The current PR is made mostly in order to help with PHP Stack related issue, and later (once https://github.com/bmewburn/vscode-intelephense/issues/465 is fixed) it should be updated with the apropriate `url` attribute value before merging

amisevsk commented on plugins/ms-vscode.php.intelephense/1.0.13/meta.yaml:
```diff
@@ -0,0 +1,14 @@
+category: Language
+firstPublicationDate: "2019-04-16"
+attributes:
+  containerImage: "eclipse/che-remote-plugin-php7:next"
```
To add this plugin to more recent fields in the registry, could you add the following `meta.yaml` to paths `./v2/plugins/redhat/php/1.0.13` and `./v3/plugins/redhat/php/1.0.13`

```yaml
apiVersion: v2
publisher: redhat
name: php
version: 1.0.13
displayName: PHP Intelephense
title: PHP Intelephense
type: VS Code extension
icon: https://www.eclipse.org/che/images/logo-eclipseche.svg
repository: https://github.com/bmewburn/vscode-intelephense
category: Language
firstPublicationDate: "2019-04-16"
extensions:
  - https://marketplace.visualstudio.com/_apis/public/gallery/publishers/bmewburn/vsextensions/vscode-intelephense-client/1.0.13/vspackage
```

The salient changes are: 
1. No `url` field; moved to `extensions`
2. `id` field deprecated; `id`s are now the combination `publisher/name/version` (updated name to `php`, publisher to `redhat` as that's what's used for other Red Hat published plugins.
3. Add `apiVersion` field as it's necessary for the v3 path.

vrubezhny replied:
@amisevsk Should I change `apiVersion: v2`  to `apiVersion: v3` in your meta.yaml in case I'm going to target the `next release`?

amisevsk replied:
No, the `v2`, `v3` in this repo is unrelated to `v2`/`v3` in `meta.yaml` spec... I know it's confusing. We were talking today about a plan for improving docs around this.

For `meta.yaml`s, at the time of writing only `v2` is supported (in the `v3` folder of the repo...).

vrubezhny commented:
@amisevsk I've moved the PHP Language Support plugin to v3/plugins and changed meta.yaml accordingly.
Please, verify.
Please do not push this plugin until we fix the extension `uri` that currently points to `https://marketplace.visualstudio.com/` and, as such, is problematic due to https://github.com/eclipse/che/issues/12840

vrubezhny commented:
A fixup is added in order to use the `vsix` archive from `che-incubator/vscode-intelephense` project instead of the marketplace. This should workaround the `VS Code marketplace rate limit error #12840` problem (eclipse/che#12840)

vrubezhny commented:
@amisevsk `setup_latest_dirs.sh` didn't work for me:
```
$ ./scripts/setup_latest_dirs.sh 
Working on directory: v3/plugins/camel-tooling/vscode-apache-camel
./scripts/setup_latest_dirs.sh: line 30: yq: command not found
```
And I wasn't able to find a source to install  `yq` tool from. So, I did it manually.
Also, I squased the commits into one.
Please verify.

amisevsk commented:
@vrubezhny Ah yes, I'll include some documentation about `yq`, especially since there are multiple programs named `yq` and only one works :)
